connect/transforms/src/test/java/org/apache/kafka/connect/transforms/HoistFieldTest.java

```diff
@@ -58,4 +59,17 @@ public void withSchema() {
         assertEquals(42, ((Struct) transformedRecord.key()).get("magic"));
     }
 
+    @Test
+    public void testSchemalessMapIsMutable() {
+        xform.configure(Collections.singletonMap("field", "magic"));
+
+        final SinkRecord record = new SinkRecord("test", 0, null, 420, null, null, 0);
+        final SinkRecord transformedRecord = xform.apply(record);
+
+        assertNull(transformedRecord.keySchema());
+        @SuppressWarnings("unchecked")
+        Map<String, Object> key = (Map<String, Object>) transformedRecord.key();
+        key.put("k", "v");
```

Comment:
nit: Probably wrap this put operation in `assertDoesNotThrow` to ensure `UnsupportedOperation` is not thrown? Something like => assertDoesNotThrow(() -> key.put("k", "v"));

Reply:
The test would anyway fail if it were to throw `UnsupportedOperation` 😄

But I guess this way it's more obvious what the test is doing 👍